src/script/view_model/CallingViewModel.ts

```diff
@@ -148,23 +156,138 @@ export class CallingViewModel {
       if (!canStart) {
         return;
       }
-      const subConversationData = conversation.isUsingMLSProtocol
-        ? await core.service!.mls?.joinConferenceSubconversation(conversation)
-        : undefined;
-
-      const keyData = subConversationData && {
-        epoch: subConversationData.subconversation.epoch,
-        secretKey: subConversationData.secretKey,
-        keyLength: subConversationData.keyLength,
-      };
 
-      const call = await this.callingRepository.startCall(conversation, callType, keyData);
+      const call = await this.callingRepository.startCall(conversation, callType);
       if (!call) {
         return;
       }
+
+      if (conversation.isUsingMLSProtocol) {
+        const unsubscribe = await subscribeToEpochUpdates(
+          conversation.qualifiedId,
+          ({epoch, keyLength, secretKey, members}) => {
+            this.callingRepository.setEpochInfo(conversation.qualifiedId, {epoch, keyLength, secretKey}, members);
+          },
+        );
+
+        callingSubscriptions.addOngoing(call.conversationId, unsubscribe);
+      }
       ring(call);
     };
 
+    const generateSubconversationMembers = async (
+      conversationId: QualifiedId,
+      subconversation: Subconversation,
+    ): Promise<SubconversationEpochInfoMember[]> => {
+      const mlsService = core.service?.mls;
+
+      if (!mlsService) {
+        throw new Error('mls service was not initialised');
+      }
+
+      const parentGroupId = await mlsService.getGroupIdFromConversationId(conversationId);
+      const memberIds = await mlsService.getClientIds(parentGroupId);
+
+      const members = memberIds.map(parentMember => {
+        const isSubconversationMember = !!subconversation.members.find(
+          ({user_id, client_id}) => user_id === parentMember.userId && client_id === parentMember.clientId,
+        );
+
+        return {
+          userid: `${parentMember.userId}@${parentMember.domain}`,
+          clientid: parentMember.clientId,
+          in_subconv: isSubconversationMember,
+        };
+      });
+
+      return members;
+    };
+
+    const getSubconversationEpochInfo = async (
+      conversationId: QualifiedId,
+      subconversation: Subconversation,
+    ): Promise<{
+      members: SubconversationEpochInfoMember[];
+      epoch: number;
+      secretKey: string;
+      keyLength: number;
+    }> => {
+      const mlsService = core.service?.mls;
+      if (!mlsService) {
+        throw new Error('mls service was not initialised');
+      }
+
+      const members = await generateSubconversationMembers(conversationId, subconversation);
+
+      const epoch = Number(await mlsService.getEpoch(subconversation.group_id));
+
+      const keyLength = 32;
+      const secretKey = await mlsService.exportSecretKey(subconversation.group_id, keyLength);
+
+      return {members, epoch, keyLength, secretKey};
+    };
+
+    const subscribeToEpochUpdates = async (
```

Comment:
Maybe we could extract all the mls-related functions to another module in order not to bloat the CallingViewModel with those details. WDYT?

Reply:
I like it, let me try excluding it 👌

Reply:
done 👌